src/intrprtr.c

```diff
@@ -2201,6 +2201,19 @@ void           IntrStringExpr (
     PushObj( string );
 }
 
+void           IntrPragma (
+    Obj               pragma )
+{
+    SKIP_IF_RETURNING();
+    SKIP_IF_IGNORING();
+    if ( STATE(IntrCoding)    > 0 ) {
+        CodePragma( pragma );
+    } else {
+        // Push a void when interpreting
+        PushVoidObj();
```

Comment:
Why?

Reply:
If you don't do it, you run into the assertion `assert( LEN_PLIST(STATE(StackObj)) == 1 );` in `IntrEnd` if you enter a pragma in plain interactive mode, i.e., in a GAP session like
```
gap> #@ foo
```
Maybe the fact that this is needed points to a different error.

Reply:
Actually, your code here is correct, I got confused between interpreter and coder. In the interpreter (like here) every expression must push an actual  value, and every statement must push a void value. Sorry for the confusion